go/api/handlers/document.go

```diff
@@ -0,0 +1,189 @@
+		fmt.Println(errFilePath)
+		context.JSON(http.StatusBadRequest, gin.H{"error": "Error saving file"})
+		return
+	}
+
+	trip, _ := handler.TripRepository.Get(tripId)
+	var document = models.Document{
+		Title:       title,
+		Description: description,
+		TripID:      trip.ID,
+		Path:        filePath,
+	}
+
+	err := handler.Repository.Create(&document)
+	if err != nil {
+		errorHandlers.HandleGormErrors(err, context)
+		return
+	}
+
+	documentResponse := responses.DocumentResponse{
+		ID:          document.ID,
+		Title:       document.Title,
+		Description: document.Description,
+		Path:        document.Path,
+		CreatedAt:   document.CreatedAt.Format(time.RFC3339),
+		UpdateAt:    document.UpdatedAt.Format(time.RFC3339),
+	}
+
+	context.JSON(200, documentResponse)
+}
+
+// @Summary      Download a document
+// @Description  Download a document
+// @Tags         document
+// @Accept       json
+// @Produce      application/octet-stream
+// @Security     BearerAuth
+// @Param        id          path      string  true  "ID of the trip"
+// @Param        documentID  path      string  true  "ID of the document"
+// @Success      200         {file}    true    "A file stream of the document"
+// @Failure      400         {object}  error
+// @Failure      401         {object}  error
+// @Failure      404         {object}  error
+// @Router       /trips/{id}/documents/{documentID}/download [get]
+func (handler *DocumentHandler) DownloadDocument(context *gin.Context) {
+	documentID := context.Param("documentID")
+
+	document, errDoc := handler.Repository.Get(documentID)
+	if errDoc != nil {
+		errorHandlers.HandleGormErrors(errDoc, context)
+		return
+	}
+
+	filepath, errFilePath := utils.GetFilePath("document", document.Path)
+	if errFilePath != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get file path"})
+		return
+	}
+	context.Header("Content-Disposition", "attachment; filename=\""+document.Path+"\"")
+	context.File(filepath)
+}
+
+// @Summary		Delete a document from trip
+// @Description	Delete a document from the trip
+// @Tags			document
+// @Accept			json
+// @Produce		json
+// @Security		BearerAuth
+// @Param			id			path		string	true	"ID of the trip"
+// @Param			documentID	path		string	true	"ID of the document"
+// @Success		204			{object}	error
+// @Failure		400			{object}	error
+// @Failure		401			{object}	error
+// @Failure		404			{object}	error
+// @Router			/trips/{id}/documents/{documentID} [delete]
+func (handler *DocumentHandler) DeleteDocumentFromTrip(context *gin.Context) {
+	tripId := context.Param("id")
+
+	documentID := context.Param("documentID")
+	if documentID == "" {
+		context.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	trip, _ := handler.TripRepository.Get(tripId)
+
+	document, errDoc := handler.Repository.Get(documentID)
+	if errDoc != nil {
+		errorHandlers.HandleGormErrors(errDoc, context)
+		return
+	}
+
+	err := handler.Repository.DeleteDocument(trip, document.ID)
```

Comment:
```suggestion
	document, errDoc := handler.Repository.Get(documentID)
	if errDoc != nil {
		errorHandlers.HandleGormErrors(errDoc, context)
		return
	}

	err := handler.Repository.Delete(document.ID)
```

Pas besoin du trip ici